src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,142 @@
+package duke;
+
+import java.util.ArrayList;
+
+import duke.tasktypes.Deadlines;
+import duke.tasktypes.Task;
+
+/**
+ * Class containing the list of tasks for Duke chatbot.
+ */
+public class TaskList {
+    
+    protected ArrayList<Task> listOfTasks;
+
+    /**
+     * Constructor to initialize a TaskList instance.
+     * @param listOfTasks An arraylist containing a list of tasks from the Duke chatbot.
+     */
+    public TaskList(ArrayList<Task> listOfTasks) {
+        this.listOfTasks = listOfTasks;
+    }
+
+    public ArrayList<Task> getListOfTasks() {
+        return this.listOfTasks;
+    }
+
+    /**
+     * Function to print out the list of tasks currently stored in the list to the user.
+     */
+    public void toRead() {
+        System.out.println("Here are the tasks in your list:");
+        for (int i = 0; i < listOfTasks.size(); i++) {
+            Integer currIndex = i + 1;
+            Task currTask = listOfTasks.get(i);
+            String toUse = currIndex.toString() + "." + currTask.toString();
+            System.out.println(toUse);
+        }
+    }
+
+    /**
+     * Deletes the task at the indicated index.
+     * @param index The index which the task is stored at.
+     * @throws DukeExceptions
+     */
+    public void deleteTask(int index) throws DukeExceptions {
+        try {
+            int indexToUse = index - 1;
+            if (indexToUse >= listOfTasks.size() || indexToUse < 0) {
+                throw new DukeExceptions("Wrong size for mark/unmark");
+            }
+            Task gettingTask = listOfTasks.remove(indexToUse);
+            String toOutput = "Noted. I've removed this task:\n  " + gettingTask.toString() + "\nNow you have " + listOfTasks.size() + " tasks in the list";
```

Comment:
This line too long. Possible to break. To not spam comments, I would not comment on other long lines but try to keep line lengths to 110 characters.